src/nvme/tree.c

```diff
@@ -1222,6 +1230,48 @@ static char *nvme_ctrl_lookup_subsystem_name(nvme_root_t r,
 	return subsys_name;
 }
 
+static char *nvme_ctrl_lookup_phy_slot(nvme_root_t r, const char *address)
+{
+	char *target_addr;
+	char *addr;
+	char *path;
+	int found = 0;
+	int ret;
+	struct dirent *entry;
+
+	DIR *slots_dir = opendir(nvme_slots_sysfs_dir);
+	if (slots_dir == NULL) {
```

Comment:
`if (!slots_dir) { ...}` is the preferred way to check for NULL pointers.

Reply:
Updated.